internal/controllers/machine/machine_controller.go

```diff
@@ -865,3 +868,10 @@ func (w writer) Write(p []byte) (n int, err error) {
 	w.logFunc(string(p))
 	return len(p), nil
 }
+
+type scope struct {
```

Comment:
Let's maybe add a bit of godoc when the fields are available. Something like that cluster & machine will be set at the beginning and infraMachine and bootstrapConfig are available after reconcileBootstrap/reconcileInfrastructure was run

Can we move this struct a bit up? Maybe directly after `reconcile`? (after because it's used in the func, not before because it's not part of the func signature of reconcile)